packages/flutter/lib/src/rendering/proxy_box.dart

```diff
@@ -1471,11 +1418,199 @@ class RenderPhysicalModel extends _RenderCustomClip<RRect> {
     }
   }
 
+  @override
+  bool hitTest(HitTestResult result, { Offset position }) {
+    if (_clipper != null) {
+      _updateClip();
+      assert(_clip != null);
+      if (!_clip.contains(position))
+        return false;
+    }
+    return super.hitTest(result, position: position);
+  }
+
   @override
   void debugFillProperties(DiagnosticPropertiesBuilder description) {
     super.debugFillProperties(description);
     description.add(new DiagnosticsProperty<BoxShape>('shape', shape));
     description.add(new DiagnosticsProperty<BorderRadius>('borderRadius', borderRadius));
+  }
+}
+
+/// Creates a physical model layer that clips its children to a [ShapeBorder].
+///
+/// A physical model layer casts a shadow based on its [elevation].
+///
+/// See also:
+///
+/// * [RenderPhysicalModel], which is optimized for rounded rectangles and
+///   circles.
+class RenderPhysicalShape extends _RenderPhysicalModelBase<Path> {
+  /// Creates an arbitrary shape clip.
+  ///
+  /// The [color], [shape], and [textDirection] are required.
+  ///
+  /// The [shape], [elevation], [color], [textDirection] and [shadowColor] must
+  /// not be null.
+  RenderPhysicalShape({
+    RenderBox child,
+    @required ShapeBorder shape,
+    double elevation: 0.0,
+    @required Color color,
```

Comment:
This is the background color for the area within the shaped border's inner path or outer path? RenderPhysicalModel.color should explain this as well.

Reply:
This is the background color for the shape itself... there is no border painted...
I guess it is somewhat ambiguous to use ShapeBorder as we only need one path for the clipping, though I'm not sure if we want another type just for that.
Let me know what you think...